src/fighter.rs

```diff
@@ -124,9 +132,88 @@ impl ActiveFighterBundle {
             .insert(Hurtbox)
             .id();
 
+        let animated_spritesheet_bundle = active_fighter_bundle.animated_spritesheet_bundle.clone();
+
         commands
             .entity(entity)
             .insert_bundle(active_fighter_bundle)
             .push_children(&[hurtbox]);
+
+        if let Some(attachment) = &fighter.attachment {
+            //Clone fighter spritesheet
+            let mut attachment_spritesheet = animated_spritesheet_bundle;
+
+            //Change what's needed
+            attachment_spritesheet.sprite_sheet.texture_atlas = attachment
+                .atlas_handle
+                .choose(&mut rand::thread_rng())
+                .unwrap()
+                .clone();
+            attachment_spritesheet.animation =
+                Animation::new(attachment.animation_fps, attachment.animations.clone());
+            attachment_spritesheet.sprite_sheet.transform =
+                Transform::from_xyz(0., fighter.spritesheet.tile_size.y as f32 * 0.3, 0.1);
+            attachment_spritesheet.sprite_sheet.sprite.anchor = bevy::sprite::Anchor::Center;
+
+            let attachment_ent = commands
+                .spawn_bundle(attachment_spritesheet)
+                .insert(Attached {
+                    position_face: false,
+                    sync_animation: true,
+                    sync_facing: true,
+                })
+                .insert(Facing::default())
+                .id();
+            commands.entity(entity).add_child(attachment_ent);
+        }
+    }
+}
+
+#[derive(Component)]
+pub struct Attached {
+    /// Syncs facing with parent facing
+    pub sync_facing: bool,
+    // Syncs animation with parent animation
+    pub sync_animation: bool,
+    /// Change position based on facing
+    pub position_face: bool,
```

Comment:
Another thing we could consider is having the actual attach point be child component of the fighter, and which gets translated in x when the fighters facing changes. Then whatever is attached to that attach point would be a child entity of that and get moved automatically.